SPIFlashLogger.class.nut

```diff
@@ -461,22 +680,30 @@ class SPIFlashLogger {
         return count+1;// TODO: why was this plus one necessary?
     }
 
+    //
+    // Initialise logger in scope of the provided logger addresses
+    //
     function _init() {
-        local best_id = 0;          // The highest id
-        local best_sec = 0;         // The sector with the highest id
-        local best_map = 0xFFFF;    // The map of the sector with the highest id
-
+        local firstSector = {"id" : 0, "sec" : 0, "map": 0xFFFF}; // The smallest id
+        local lastSector = {"id" : 0, "sec" : 0, "map": 0xFFFF};// The highest id
         // Read all the metadata
         _enable();
         for (local sector = 0; sector < _sectors; sector++) {
             // Hunt for the highest id and its map
             local meta = _getSectorMetadata(sector);
 
             if (meta.id > 0) {
-                if (meta.id > best_id) {
-                    best_sec = sector;
-                    best_id = meta.id;
-                    best_map = meta.map;
+                // identify last sector
+                if (meta.id > lastSector.id) {
+                    lastSector.sec = sector;
+                    lastSector.id = meta.id;
+                    lastSector.map = meta.map;
+                }
+                // identify first sector
+                if (firstSector.id == 0 || meta.id < firstSector.id) {
```

Comment:
Again here would be just a comparison with null instead of `.id == 0`

Reply:
Why null ?  If we have an empty sector (clean) then and id == 0, otherewise id != 0

For dirty sector an .id = 0xFFFFFFFFF